File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        hs = set(nums)
        largest = 0

        for num in hs:
            if num-1 not in hs:
                curr = num
                while curr in hs:
                    curr += 1
                largest = max(largest, curr-num)
        
        return largest

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums == null || nums.length == 0) return 0;
        int largest = 0;
        Set<Integer> numsSet = new HashSet<>();
        for (int num: nums) numsSet.add(num);

        for (int num: numsSet) {
            if (!numsSet.contains(num-1)) {
                int curr = num;
                while (numsSet.contains(curr)) {
                    curr += 1;
                }
                largest = Math.max(largest, curr-num);
            }
        }

        return largest;
    }
}